### PR TITLE
Scraping data

### PR DESCRIPTION
**Pull request by SamEGil00**
Scraped all 163 majors + minors, current data includes program name + school.

Started a script to transform course data json into format that is better suited for seeding into django db

### CODE REVIEW

**Review by joshhn — APPROVED** [review]
lgtm

File: backend/data-scraper/washu_courses_data_fixed.json [hunk 1]
```diff
@@ -0,0 +1,1042 @@
...
+    },
+    {
+        "course_code": "CSE 131",
+        "course_name": "Introduction",
+        "dept_code": "E81",
+        "units": "Credit",
```
Comment (ngoctramvu) [inline backend/data-scraper/washu_courses_data_fixed.json:22]: "units" are missing the actual credits value

File: backend/data-scraper/washu_major_data.json
Comment (ngoctramvu) [inline backend/data-scraper/washu_major_data.json:1]: Not sure what your intention is with the major.json and minor.json. These data is not following the format specified by the schema.

File: frontend/src/routes/LoginPage.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ export default function LoginPage() {
 
   const checkUser = () => {
     // TODO: call from API
-    setUserExisted(true)
+    setUserExisted(false)
```
Comment (ngoctramvu) [inline frontend/src/routes/LoginPage.tsx:19]: Why is this changed? This is not related to the database

File: backend/data-scraper/course fix.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    """Export data to json file."""
+    with open(
+        "C:/Users/sameg/OneDrive/Documents/GitHub/FL24-project-group-9/backend/data-scraper/washu_courses_data_fixed.json",
+        "w+",
+        encoding="utf-8",
+        ) as output_file:
+        output_file.write(json.dumps(courses_list, indent=4))
+        print(f"{len(courses_list)} courses exported to file")
+
+df = pd.read_json("C:/Users/sameg/OneDrive/Documents/GitHub/FL24-project-group-9/backend/data-scraper/washu_courses_data.json")
```
Comment (ngoctramvu) [inline backend/data-scraper/course fix.py:14]: this should not be read from your local storage

File: backend/data-scraper/course fix.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import json
+import pandas as pd
+
+def export_data(courses_list: list):
+    """Export data to json file."""
+    with open(
+        "C:/Users/sameg/OneDrive/Documents/GitHub/FL24-project-group-9/backend/data-scraper/washu_courses_data_fixed.json",
```
Comment (ngoctramvu) [inline backend/data-scraper/course fix.py:7]: same as below, should not be read from local file path

File: backend/data-scraper/course fix.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    for x in range(len(df)):
+        course_info = {}
+        concat_title = df['title'][x].split(' ', 4)
+        concat_units = df['credits'][x].split(' ', 3)
+        dept_code = concat_title[0]
+        course_code = concat_title[1] + " " + concat_title[2]
+        course_name = concat_title[3]
+        course_url = df['link'][x]
+        if df['credits'][x].split(' ', 3)[1]
+        course_units = df['credits'][x].split(' ', 3)[1]
```
Comment (ngoctramvu) [inline backend/data-scraper/course fix.py:27]: this if statement is not indented. What should be inside the if block?

File: backend/data-scraper/program scraper.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    def __init__(self, base_url: str):
+        """Initialize attributes of class."""
+        self.url = base_url
+
+    def export_data(self, programs_list: list, pType):
+        """Export data to json file."""
+        if pType.lower() == "major":
+            with open(
+                "C:/Users/sameg/OneDrive/Documents/GitHub/FL24-project-group-9/backend/data-scraper/washu_major_data.json",
```
Comment (ngoctramvu) [inline backend/data-scraper/program scraper.py:35]: please fix file path

File: backend/data-scraper/program scraper.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                "w+",
+                encoding="utf-8",
+            ) as output_file:
+                output_file.write(json.dumps(programs_list, indent=4))
+                print(f"{len(programs_list)} courses exported to file")
+        elif pType.lower() == "minor":
+            with open(
+                "C:/Users/sameg/OneDrive/Documents/GitHub/FL24-project-group-9/backend/data-scraper/washu_minor_data.json",
```
Comment (ngoctramvu) [inline backend/data-scraper/program scraper.py:43]: please fix file path

File: backend/data-scraper/program scraper.py [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+                encoding="utf-8",
+            ) as output_file:
+                output_file.write(json.dumps(programs_list, indent=4))
+                print(f"{len(programs_list)} courses exported to file")
+
+    def create_programs_list(self, pType) -> list:
+
+        """Return the list of majors and minors from the page."""
+        print("Creating majors and minors list...")
+        # self.driver.get(self.url)
```
Comment (ngoctramvu) [inline backend/data-scraper/program scraper.py:54]: remove unused code comments

File: backend/data-scraper/course fix.py (hunk 4 above)
Comment (joshhn) [reply backend/data-scraper/course fix.py:7]: The file name should not contain the space, should be `course_fix.py`.
